src/OAuth2Form.tsx

```diff
@@ -15,16 +14,30 @@ import {
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { OAuthConsentGrantedResponse } from 'synapse-react-client/dist/utils/synapseTypes'
 import { getURLParam, getStateParam, handleErrorRedirect } from './URLUtils'
+import { SynapseContextType } from 'synapse-react-client/dist/utils/SynapseContext'
 
 // can override endpoints as https://repo-staging.prod.sagebase.org/ and https://staging.synapse.org for staging
 
 const isStaging:boolean = window.location.hostname.includes('staging');
+const isDev:boolean = window.location.hostname.includes('dev');
+
+const prodConfig = {
+    REPO: 'https://repo-prod.prod.sagebase.org/',
+    PORTAL: 'https://www.synapse.org/',
+};
+
+const stagingConfig = {
+    REPO: 'https://repo-staging.prod.sagebase.org/',
+    PORTAL: 'https://staging.synapse.org/',
+};
+
+const devConfig = {
+    REPO: 'https://repo-dev.dev.sagebase.org/',
+    PORTAL: 'https://portal-dev.dev.sagebase.org/',
+};
 
 (window as any).SRC = {
-    OVERRIDE_ENDPOINT_CONFIG: {
-        REPO: isStaging ? 'https://repo-staging.prod.sagebase.org/' : 'https://repo-prod.prod.sagebase.org/',
-        PORTAL: isStaging ? 'https://staging.synapse.org/' : 'https://www.synapse.org/',
-    }
+    OVERRIDE_ENDPOINT_CONFIG: isStaging ? stagingConfig : isDev ? devConfig : prodConfig
```

Comment:
This is the real fix for PORTALS-2082, and we need it to be accessed from a url that contains "dev" in order to verify that these new endpoints are used.
If "staging" is present (will be in "staging-signin.synapse.org") then use the staging endpoints.  Otherwise if "dev" is present (recommending you use something like "dev-signin.synapse.org") then use the dev endpoints.  Otherwise use the production endpoints.